src/TreeMap.js

```diff
@@ -129,11 +129,12 @@ class TreeMap extends React.Component {
     this._data = _.cloneDeep(this.props.data);
   }
   componentWillReceiveProps(newProps){
-    const {width, height} = this.props;
+    const {width, height, data, sticky} = this.props;
 
-    //if height or width change, re-initialize the layout function
-    if(width != newProps.width || height != newProps.height)
+    //if height, width, or the data changes, or if the treemap is not sticky, re-initialize the layout function
+    if(!sticky || width != newProps.width || height != newProps.height || JSON.stringify(data) != JSON.stringify(newProps.data))
```

Comment:
I would just do a `===` check here, since most of the library assumes that you will be following immutability rules (changing reference if data changes), and `JSON.stringify` will be slow if data is large.

In general, if you have to do deep equality checks, I think `_.isEqual` will always be faster than `JSON.stringify` since it doesn't have to do string manipulation. But I don't think it should be necessary here.